File: AptitudeAce-Backend/src/main/java/com/aptitude/service/QuestionService.java
package com.aptitude.service;

import java.util.Set;

import com.aptitude.entity.Question;
import com.aptitude.entity.Quiz;

public interface QuestionService {

    public Question addQuestion(Question question);

    public Question updateQuestion(Question question);

    public Set<Question> getQuestions();

    public Question getQuestion(Long questionId);

    public Set<Question> getQuestionsOfQuiz(Quiz quiz);

    public void deleteQuestion(Long quesId);

    public Question get(Long questionsId);

}

File: AptitudeAce-Frontend/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {
 
  constructor(private http: HttpClient) { }

  private baseUrl = 'http://localhost:8888'; //Spring boot url

  getQuestionsOfQuiz(qid: any) {
    return this.http.get(`${this.baseUrl}/question/quiz/${qid}`);
  }

  addQuestion(question:any) {
    return this.http.post(`${this.baseUrl}/question/`, question);
  }

  deleteQuestion(quesId: any) {
    return this.http.delete(`${this.baseUrl}/question/${quesId}`);
  }

  evaluateQuiz(questions:any) {
    return this.http.post(`${this.baseUrl}/question/eval-quiz`, questions);
  }
}
